extensions/objects/templates/FilepathEditor.js

```diff
@@ -31,109 +31,118 @@ oppia.directive('filepathEditor', [
       scope: true,
       template: '<div ng-include="getTemplateUrl()"></div>',
       controller: ['$scope', function($scope) {
-        // Reset the component each time the value changes (e.g. if this is part
-        // of an editable list).
+        // Reset the component each time the value changes
+        // (e.g. if this is part of an editable list).
         $scope.$watch('$parent.value', function(newValue) {
-          $scope.localValue = {
-            label: newValue || ''
-          };
-          $scope.imageUploaderIsActive = false;
+          if (newValue) {
+            $scope.setSavedImageUrl(newValue, false);
+          }
         });
 
-        $scope.explorationId = explorationContextService.getExplorationId();
+        $scope.validate = function(state) {
+          return state.state === 'saved' &&
```

Comment:
`state` is confusing. Maybe, at the outset, define:

```
  $scope.data = {
    mode: MODE_EMPTY,
    savedUrl: ''
  };
```

and use that as your "state" object. That way, its fields are also fixed and you don't even have to do the check for whether data.state exists.